book/contributions/rec_pubchem_ptable/recipes/rec_pubchem_ptable.ipynb

```diff
@@ -0,0 +1,527 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a5e39374-4aeb-4419-b4a6-292a2e748c11",
+   "metadata": {},
+   "source": [
+    "# Accessing the Chemical Elements Data from PubChem"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0707cd8c-6365-4ef2-8086-a0d5d7c6e57a",
+   "metadata": {},
+   "source": [
+    "PubChem provides the Periodic Table (https://pubchem.ncbi.nlm.nih.gov/periodic-table/) and Element Pages to help users navigate the abundant chemical element data available within PubChem.  More detailed information on these services is provided in the following paper:\n",
+    "\n",
+    ">**PubChem Periodic Table and Element Pages: Improving Access to Information on Chemical Elements from Authoritative Sources**  \n",
+    ">Kim et al., Chemistry Teacher International, 2021, 3(1):57-65.  \n",
+    ">DOI: [10.1515/cti-2020-0006](https://doi.org/10.1515/cti-2020-0006)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a98076bd-268c-4367-89ab-d628d633a671",
+   "metadata": {},
+   "source": [
+    "The data underlying the PubChem Periodic Table and Element Pages are freely available for download.  This Jupyter notebook demonstrates how to download the properties of chemical elements and visualize the periodicity of those properties.  The data used in this notebook can be downloaded from the PubChem Periodic Table (https://pubchem.ncbi.nlm.nih.gov/periodic-table/), by clicking the **DOWNLOAD** button and then the **CSV** button."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aea2fa0b-e380-418a-920b-7ee55fc41121",
+   "metadata": {},
+   "source": [
+    "![PubChem-Periodic-Table](../images/rec_pubchem_ptable_fig1.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e38ed4d-acf1-4aeb-ae83-d0ecec97f4e7",
+   "metadata": {},
+   "source": [
+    "Alternatively, the data can also be downloaded using a python script, as demonstrated below.  Here, we use a python data analysis library called pandas.  Using pandas, we can directly load the data from the web into a data frame."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d2e94951-02ec-42ad-a862-3575a814fc16",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import pandas as pd"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d4cb5d20-e006-4083-8af0-6852861af44c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = \"https://pubchem.ncbi.nlm.nih.gov/rest/pug/periodictable/CSV\"\n",
+    "df = pd.read_csv(url)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "edcb00ba-f5a3-45f5-ae0c-29760b045d3c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df"
```

Comment:
Style comment: Is it necessary to show whole dataframe here? What is the purpose and does it need describing?

Reply:
I guess in Jupyter it will auto display only the top 5 and bottom 5 - does this need making explicit in the comment?

Reply:
In my view, whenever students load data from somewhere, they should check if the data are loaded correctly.  The line with "df" is added for this purpose.  In Jupyter, this line indeed shows the first 5 and last 5 lines of the loaded data.  The next two lines (with df.shape and df.columns) are also added for this purpose. 

In a real-world exploratory data analysis, more thorough data integrity/quality checking should be done, but I added only three lines for data integrity/quality check (as a minimum measure).  I thought about adding a few sentences about this topic in the notebook, but did not actually do it because I felt it would be beyond the scope of this notebook.

Reply:
@Smund27 Can you resolve this tomorrow?  I would like to merge tomorrow and get this recipe up on the website.